api/src/middleware/programsRouter.ts

```diff
@@ -18,4 +23,84 @@ programsRouter.get('/', async (req, res, next) => {
   );
 });
 
+programsRouter.get(
+  '/activityStatus/:programId/:activityId',
+  async (req, res, next) => {
+    const { programId, activityId } = req.params;
+    const { principalId } = req.session;
+
+    const programIdNum = Number(programId);
+    const activityIdNum = Number(activityId);
+
+    if (!Number.isInteger(programIdNum) || programIdNum < 1) {
+      next(new BadRequestError(`"${programIdNum}" is not a valid program id.`));
+      return;
+    }
+
+    if (!Number.isInteger(activityIdNum) || activityIdNum < 1) {
+      next(new BadRequestError(`"${activityId}" is not a valid activity id.`));
+      return;
+    }
+
+    let activityCompletionStatus;
+
+    try {
+      activityCompletionStatus = await getParticipantActivityCompletion(
+        principalId,
+        programIdNum,
+        activityIdNum
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+
+    res.json(
+      itemEnvelope({
+        programId: programIdNum,
+        activityId: activityIdNum,
+        completed: activityCompletionStatus.completed,
+      })
+    );
+  }
+);
+
+programsRouter.put(
+  '/activityStatus/:programId/:activityId',
+  async (req, res, next) => {
+    const { programId, activityId } = req.params;
+    const { completed } = req.body;
+    const { principalId } = req.session;
+
+    const programIdNum = Number(programId);
+    const activityIdNum = Number(activityId);
+
+    if (!Number.isInteger(programIdNum) || programIdNum < 1) {
+      next(new BadRequestError(`"${programIdNum}" is not a valid program id.`));
+      return;
+    }
+
+    if (!Number.isInteger(activityIdNum) || activityIdNum < 1) {
+      next(new BadRequestError(`"${activityId}" is not a valid activity id.`));
+      return;
+    }
+
+    let updatedParticipantActivity;
+
+    try {
+      updatedParticipantActivity = await setParticipantActivityCompletion(
+        principalId,
+        programIdNum,
+        activityIdNum,
+        completed
+      );
+    } catch (error) {
+      next(error);
+      return;
+    }
+
+    res.status(201).json(itemEnvelope(updatedParticipantActivity));
+  }
```

Comment:
Also Phoebe updated this service function.
I'm not sure did she update her branch or main branch.